From 192bfd9101ad931fb077fbb11f474c9ad98c8fb3 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Fri, 18 Aug 2017 12:06:18 +0800
Subject: [PATCH 12/25] DisplayManagerService: disable display blanking on
 suspend

A workaround for black screen that SurfaceFlinger never recovers from.
The kernel does this anyway.

Re-worked from the original kitkat-x86 patch of Kyle Evans.
---
 .../com/android/server/display/DisplayManagerService.java     | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/display/DisplayManagerService.java b/services/core/java/com/android/server/display/DisplayManagerService.java
index eea22cfdd96..a14e87a6604 100644
--- a/services/core/java/com/android/server/display/DisplayManagerService.java
+++ b/services/core/java/com/android/server/display/DisplayManagerService.java
@@ -540,7 +540,9 @@ public final class DisplayManagerService extends SystemService {
                             + ", brightness=" + brightness + ")");
                     mGlobalDisplayState = state;
                     mGlobalDisplayBrightness = brightness;
-                    applyGlobalDisplayStateLocked(mTempDisplayStateWorkQueue);
+                    if (state != Display.STATE_OFF) {
+                        applyGlobalDisplayStateLocked(mTempDisplayStateWorkQueue);
+                    }
                 }
 
                 // Setting the display power state can take hundreds of milliseconds
-- 
2.17.1

